**Q1. Write a Python function that takes in two arrays of data and calculates the F-value for a variance ratio
test. The function should return the F-value and the corresponding p-value for the test.**

**Ans:**

In [2]:
import numpy as np
import scipy.stats as stat

worker1= [18, 19, 22, 25, 27, 28, 41, 45, 51, 55]
worker2= [14, 15, 15, 17, 18, 22, 25, 25, 27, 34]

x = np.array(worker1)
y = np.array(worker2)

def f_and_p(x,y):
    f = np.var(x)/np.var(y)
    x = len(x) - 1
    y = len(y) - 1
    p_value = 1-stat.f.cdf(f,x,y)
    print(f"F-value: {f} \np-value: {p_value}")

f_and_p(x,y)

F-value: 4.387122002085506 
p-value: 0.01912653593238578


**Q2. Given a significance level of 0.05 and the degrees of freedom for the numerator and denominator of an
F-distribution, write a Python function that returns the critical F-value for a two-tailed test.**

**Ans:**

In [3]:
significance_level = 0.05
stat.t.ppf(q=1-.05/2,df=49)

2.009575234489209

**Q3. Write a Python program that generates random samples from two normal distributions with known variances and uses an F-test to determine if the variances are equal. The program should output the F-value, degrees of freedom, and p-value for the test.**

**Ans:**

In [4]:
import numpy as np
from scipy.stats import f
np.random.seed(42)

sample1 = np.random.normal(loc=0,scale=1,size=100)
sample2 = np.random.normal(loc=0,scale=1.5,size=100)

# Calculate variance
var1 = np.var(sample1,ddof=1)
var2 = np.var(sample2,ddof=1)

F_test = var1/var2

df1 = len(sample1) - 1
df2 = len(sample2) - 2
df = df1  +  df2

p_value = f.cdf(F_test,df1,df2)

print("The F-value: ",F_test)
print("The Degree of freedom: ",df)
print("The P-value: ",p_value)

The F-value:  0.4030463392763856
The Degree of freedom:  197
The P-value:  4.763749192132515e-06


**Q4.The variances of two populations are known to be 10 and 15. A sample of 12 observations is taken from each population. Conduct an F-test at the 5% significance level to determine if the variances are significantly different.**

**Ans:**

To conduct an F-test at the 5% significance level to determine if the variances of two populations with known variances of 10 and 15 are significantly different, we can use the following steps:

1. Define the null and alternative hypotheses:

    * Null hypothesis: The variances of the two populations are equal.

    * Alternative hypothesis: The variances of the two populations are not equal.
    

2. Determine the significance level (α) and the degrees of freedom (df) for the F-distribution.

    Since we are conducting the F-test at a 5% significance level, α = 0.05. The degrees of freedom for the numerator and denominator are given by:

        df numerator = n1 - 1 = 12 - 1 = 11
    
        df denominator = n2 - 1 = 12 - 1 = 11
    

3. Calculate the F-statistic:

    F = S1^2 / S2^2

    where S1^2 and S2^2 are the sample variances of the two populations.

    Since the variances of the two populations are known, we can use them as the sample variances. Thus, we have:

        F = 10 / 15 = 0.67
    

4. Determine the critical value for the F-distribution:

    Using an F-table with α = 0.05 and df numerator = 11 and df denominator = 11, we find the critical value to be 2.54.
    

5. Compare the F-statistic to the critical value:

    Since the F-statistic of 0.67 is less than the critical value of 2.54, we fail to reject the null hypothesis. We do not have enough evidence to conclude that the variances of the two populations are significantly different at the 5% significance level.

Therefore, we can conclude that there is not enough evidence to suggest that the variances of the two populations are significantly different at a 5% significance level.

**Q5. A manufacturer claims that the variance of the diameter of a certain product is 0.005. A sample of 25 products is taken, and the sample variance is found to be 0.006. Conduct an F-test at the 1% significance level to determine if the claim is justified.**

**Ans:**

To conduct an F-test at the 1% significance level to determine if the claim that the variance of the diameter of a certain product is 0.005 is justified, we can use the following steps:


1. Define the null and alternative hypotheses:

    * Null hypothesis: The variance of the diameter of the product is equal to 0.005.

    * Alternative hypothesis: The variance of the diameter of the product is greater than 0.005.


2. Determine the significance level (α) and the degrees of freedom (df) for the F-distribution.

    Since we are conducting the F-test at a 1% significance level, α = 0.01. The degrees of freedom for the numerator and denominator are given by:

        df numerator = n - 1 = 25 - 1 = 24

        df denominator = N - n = N - 25

        where N is the total number of products.


3. Calculate the F-statistic:

    F = S^2 / σ^2

    where S^2 is the sample variance and σ^2 is the population variance.

    We have S^2 = 0.006 and σ^2 = 0.005. Thus, we have:

        F = 0.006 / 0.005 = 1.2


4. Determine the critical value for the F-distribution:

    Using an F-table with α = 0.01 and df numerator = 24 and df denominator = N - 25, we find the critical value to be 3.98.


5. Compare the F-statistic to the critical value:

    Since the F-statistic of 1.2 is less than the critical value of 3.98, we fail to reject the null hypothesis. We do not have enough evidence to conclude that the variance of the diameter of the product is significantly greater than 0.005 at the 1% significance level.


Therefore, we can conclude that there is not enough evidence to suggest that the manufacturer's claim that the variance of the diameter of the product is 0.005 is unjustified at a 1% significance level.

**Q6. Write a Python function that takes in the degrees of freedom for the numerator and denominator of an F-distribution and calculates the mean and variance of the distribution. The function should return the mean and variance as a tuple.**

**Ans:**

In [2]:
import numpy as np

def f_distribution_stats(df_num, df_den):
    mean = df_den / (df_den - 2)
    var = (2 * (df_den**2) * (df_num + df_den - 2)) / ((df_num * (df_den - 2)**2 * (df_den - 4)))
    return mean, var

mean, var = f_distribution_stats(5, 10)
print("Mean: ", mean)
print("Variance: ", var)

Mean:  1.25
Variance:  1.3541666666666667


**Q7. A random sample of 10 measurements is taken from a normal population with unknown variance. The sample variance is found to be 25. Another random sample of 15 measurements is taken from another normal population with unknown variance, and the sample variance is found to be 20. Conduct an F-test at the 10% significance level to determine if the variances are significantly different.**

**Ans:**

To conduct an F-test at the 10% significance level to determine if the variances of two normal populations are significantly different, we can use the following steps:


1. Define the null and alternative hypotheses:

    * Null hypothesis: The variances of the two populations are equal.
    * Alternative hypothesis: The variances of the two populations are not equal.
    

2. Determine the significance level (α) and the degrees of freedom (df) for the F-distribution.

    Since we are conducting the F-test at a 10% significance level, α = 0.1. The degrees of freedom for the numerator and denominator are given by:

        df numerator = n1 - 1 = 10 - 1 = 9
        
        df denominator = n2 - 1 = 15 - 1 = 14

        where n1 and n2 are the sample sizes.
        

3. Calculate the F-statistic:

    F = s1^2 / s2^2

    where s1^2 and s2^2 are the sample variances.

    We have s1^2 = 25 and s2^2 = 20. Thus, we have:

    F = 25 / 20 = 1.25
    

4. Determine the critical values for the F-distribution:

    Using an F-table with α = 0.1, df numerator = 9 and df denominator = 14, we find the critical values to be 0.284 and 3.051.
    

5. Compare the F-statistic to the critical values:

    Since the F-statistic of 1.25 falls between the critical values of 0.284 and 3.051, we fail to reject the null hypothesis. We do not have enough evidence to conclude that the variances of the two populations are significantly different at the 10% significance level.
    

Therefore, we can conclude that there is not enough evidence to suggest that the variances of the two normal populations are significantly different at a 10% significance level.

**Q8. The following data represent the waiting times in minutes at two different restaurants on a Saturday night: Restaurant A: 24, 25, 28, 23, 22, 20, 27; 
Restaurant B: 31, 33, 35, 30, 32, 36. 
Conduct an F-test at the 5% significance level to determine if the variances are significantly different.**

**Ans:**

To conduct an F-test at the 5% significance level to determine if the variances of two populations are significantly different, we can use the following steps:


1. Define the null and alternative hypotheses:

    * Null hypothesis: The variances of the two populations are equal.
    * Alternative hypothesis: The variances of the two populations are not equal.
    

2. Determine the significance level (α) and the degrees of freedom (df) for the F-distribution.

    Since we are conducting the F-test at a 5% significance level, α = 0.05. The degrees of freedom for the numerator and denominator are given by:

        df numerator = n1 - 1 = 7 - 1 = 6
    
        df denominator = n2 - 1 = 6 - 1 = 5

        where n1 and n2 are the sample sizes.
        

3. Calculate the sample variances:

    We have n1 = 7 and n2 = 6. The sample variances for Restaurant A and Restaurant B are:

    s1^2 = ((24-25.7)^2 + (25-25.7)^2 + (28-25.7)^2 + (23-25.7)^2 + (22-25.7)^2 + (20-25.7)^2 + (27-25.7)^2) / (7-1) = 8.86

    s2^2 = ((31-32.83)^2 + (33-32.83)^2 + (35-32.83)^2 + (30-32.83)^2 + (32-32.83)^2 + (36-32.83)^2) / (6-1) = 5.98
    

4. Calculate the F-statistic:

    F = s1^2 / s2^2

    We have F = 8.86 / 5.98 = 1.48
    

5. Determine the critical values for the F-distribution:

    Using an F-table with α = 0.05, df numerator = 6 and df denominator = 5, we find the critical values to be 0.217 and 5.385.
    

6. Compare the F-statistic to the critical values:

    Since the F-statistic of 1.48 falls between the critical values of 0.217 and 5.385, we fail to reject the null hypothesis. We do not have enough evidence to conclude that the variances of the two populations are significantly different at the 5% significance level.
    

Therefore, we can conclude that there is not enough evidence to suggest that the variances of the waiting times at Restaurant A and Restaurant B are significantly different at a 5% significance level.

**Q9. The following data represent the test scores of two groups of students: Group A: 80, 85, 90, 92, 87, 83; Group B: 75, 78, 82, 79, 81, 84. Conduct an F-test at the 1% significance level to determine if the variances are significantly different.**

**Ans:**

To conduct an F-test for the equality of variances between the two groups, we can use the following steps:


1. State the null and alternative hypotheses

    * Null hypothesis: The variances of the two groups are equal.

    * Alternative hypothesis: The variances of the two groups are not equal.
    

2. Set the level of significance

    Let's set the level of significance to α = 0.01.
    

3. Calculate the test statistic

    We can calculate the test statistic using the formula:

        F = s1^2 / s2^2

        where s1^2 and s2^2 are the sample variances for groups A and B, respectively.

    s1^2 = ((80-86.17)^2 + (85-86.17)^2 + (90-86.17)^2 + (92-86.17)^2 + (87-86.17)^2 + (83-86.17)^2) / 5 = 21.97

    s2^2 = ((75-80)^2 + (78-80)^2 + (82-80)^2 + (79-80)^2 + (81-80)^2 + (84-80)^2) / 5 = 7.2

    F = 21.97 / 7.2 = 3.07
    

4. Determine the critical value

    The critical value for an F-test with (n1-1) and (n2-1) degrees of freedom at α = 0.01 can be found using a table or a calculator. For this problem, the critical value is 7.32.
    

5. Make a decision

    Since the calculated F-value (3.07) is less than the critical value (7.32), we fail to reject the null hypothesis. We do not have enough evidence to conclude that the variances of the two groups are significantly different at the 1% level of significance.
    

Therefore, we can conclude that there is not enough evidence to suggest that the variances of the two groups are significantly different.